ios/app/brave_core_main.mm

```diff
@@ -232,6 +236,13 @@ - (instancetype)initWithUserAgent:(NSString*)userAgent
         GetApplicationContext()->GetComponentUpdateService();
     DCHECK(cus);
 
+    // Create and start the local data file service and component installer
+    brave::URLSanitizerService* urlSanitizer =
+        brave::URLSanitizerServiceFactory::GetServiceForState(
```

Comment:
Are you sure this is not returning `nil`? I don't see `URLSanitizerServiceFactory` registered anywhere in `EnsureBrowserStateKeyedServiceFactoriesBuilt`.

That means, this service factory will return `nullptr` and this would be undefined behaviour using it.

Reply:
no it works. but its possible I'm still doing something wrong. I don't really understand these factories all that well.

Reply:
This factory should be built in this file like the rest: https://github.com/brave/brave-core/blob/master/ios/browser/browser_state/brave_browser_state_keyed_service_factories.cc

Reply:
It might work right now, but there's no guarantee that it will work in the future unless you register it yourself.
If for some reason it is registered for you in Chromium, they may remove that registration in the future, so you should register it yourself using the above function linked by Kyle.

Reply:
Actually the registration is primarily for services that are eagerly loaded. `GetInstance` is called anyway the first time you access the factory. The problem with this pattern is that you are eagerly loading it even though it could be lazy loaded. It would be better to lazy load this when you need to call methods on it rather than create an instance of the service at init time

Reply:
It's not clear to me if I'm doing what is expected here or if there is some difference of opinion about how we should approach this. But FYI, I eagerly load the component because I want to ensure I use this component is ready (downloaded) etc before I use it rather than downloading it lazily.

Reply:
@cuba The starting of the local file data service should happen eagerly, but the url sanitizer service doesn't necessarily need to start eagerly. Desktop starts it in [`PostBrowserStart`](https://github.com/brave/brave-core/blob/0e1a32ee0b1af6804cd77e1c05e8add48176cfc2/browser/brave_browser_process_impl.cc#L241). We don't have an exact replica of that, but you could use BraveCoreMain where we for instance register other components with the component updater

Reply:
This code is removed. There is no need to start the url sanitizer service anymore. Everything works now.

Reply:
@Brandon-T can we resolve this comment?